docs/writing-docs/autodocs.md

```diff
@@ -55,16 +55,41 @@ By default, Storybook offers zero-config support for documentation and automatic
+| Option        | Description                                                                                                                          |
+| ------------- | ------------------------------------------------------------------------------------------------------------------------------------ |
+| `disable`     | Toggles support for all documentation pages <br/> `docs: { disable:true }`                                                           |
+| `autodocs`    | Configures auto-generated documentation pages. Available options: `true`, `false`,`tags` (default) <br/> `docs: { autodocs: false }` |
+| `defaultName` | Renames the auto-generated documentation page<br/> `docs: { defaultName: 'Documentation' }`                                          |
 
 ### Write a custom template
 
-To replace the default documentation template used by Storybook, you can create your own template written in MDX and update your UI configuration file (i.e., `.storybook/preview.js`) to point to it.
+To replace the default documentation template used by Storybook, you can extend your UI configuration file (i.e., `.storybook/preview.js`) and introduce a `docs` [parameter](../writing-stories/parameters.md#global-parameters). This parameter accepts a `page` function that returns a React component, which you can use to generate the required template. For example:
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+    'common/storybook-preview-auto-docs-custom-template-function.js.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+<div class="aside">
+💡 Internally, Storybook uses a similar implementation to generate the default template. See the Doc Blocks API reference if you want to learn more about how Doc Blocks work.
+
+</div>
+
+Going over the code snippet in more detail. When Storybook starts up, it will override the default template with the custom one composed of the following:
+
+1. A header with the component's metadata retrieved by the `Title`, `Subtitle`, and `Description` Doc Blocks.
+2. The first story defined in the file via the `Primary` Doc Block with a handy set of UI controls to zoom in and out of the component.
+3. An interactive table with all the relevant [`args`](../writing-stories/args.md) and [`argTypes`](../api/argtypes.md) defined in the story via the `Controls` Doc Block.
+4. A overview of the remaining stories via the `Stories` Doc Block.
+
+#### With MDX
+
+However, you're not restricted to defining a function to generate the documentation template. The `docs` [parameter](../writing-stories/parameters.md#global-parameters) also accepts a reference to a template written in MDX, but this comes with a caveat, depending on where you place the file in your project, it may inadvertently get picked up by Storybook's auto-generated documentation. We recommend placing the file in a custom directory outside the `src` directory to avoid this issue. For example:
```

Comment:
A few notes here:

1. Once available, "`docs` parameter" should link to the API reference, not generic parameters documentation.
2. I'm pretty sure I've seen @tmeasday mention that you must provide a React component now and can no longer use MDX for this.
3. Not all projects use a `src` directory. Better to reference the `stories` configuration and recommend placing it somewhere that won't match?
    - This is moot if I'm correct about (2)

Reply:
I left the `docs` parameter as a placeholder. Once the API is out, this will be updated to point at it. And to this point
> I'm pretty sure I've seen @tmeasday mention that you must provide a React component now and can no longer use MDX for this.
This is an escape hatch. You can write your custom templates and place them outside a directory that is not picked by the stories glob pattern. I left this intentionally to avoid having the community reach out to us in our communication channels and ask why they're forced to write React when they work with  Vue, Angular, etc. The directory choice (i.e., `src`) was intentional, as most framework generators already set it up. I'm aware of the fact that Web Components and a monorepo infrastructure work differently.

Reply:
> This is an escape hatch. You can write your custom templates and place them outside a directory that is not picked by the stories glob pattern.

I'm confused. Are you documenting how to override the docs template or are you illustrating an alternative approach where you don't use AutoDocs at all and instead provide a custom MDX file?

Reply:
> I'm pretty sure I've seen @tmeasday mention that you must provide a React component now and can no longer use MDX for this.

I'm sure you are correct about this, but I (nor @shilman) remember the exact reason for that, and the proof is in the pudding-it indeed does work!

> This is an escape hatch. You can write your custom templates and place them outside a directory that is not picked by the stories glob pattern.

There's one more detail -- you can use the `isTemplate` prop to `Meta` (which we should document too, I didn't because I thought it didn't work 🤦 ) to stop a `.mdx` file from being indexed:

```mdx
import { Meta, Title } from '@storybook/blocks';

## This is an autodocs template

<Meta isTemplate />

<Title />
```

Reply:
I added some clarification on it with the latest commit.

Reply:
I'll add `isTemplate` to the API reference. 👍